simulator/virtual_machine.go

```diff
@@ -1475,7 +1475,7 @@ func (vm *VirtualMachine) CloneVMTask(ctx *Context, req *types.CloneVM_Task) soa
 
 		ref := ctask.Info.Result.(types.ManagedObjectReference)
 		clone := Map.Get(ref).(*VirtualMachine)
-		clone.configureDevices(&types.VirtualMachineConfigSpec{DeviceChange: req.Spec.Location.DeviceChange})
+		clone.configureDevices(&types.VirtualMachineConfigSpec{DeviceChange: req.Spec.Config.DeviceChange})
```

Comment:
Note that you'll need to check `if req.Spec.Config != nil` as it is an optional field.

Reply:
Applied this change. All tests pass now on my local machine, and setting mac address on clone from pyvmomi works as well.

Thank you!